### PR TITLE
Reorder log entries

### LINKED ISSUE #27397: Reorder the elements of the log entries to make the logs easier to read
This is a current log entry:
```
{"reqId":"Q71WK4Sp6B03QqpOoxit","remoteAddr":"173.18.243.1","app":"core","message":"Generating preview for \"\/xxxx\/yyyy\/zzzz\/0000.txt\" with \"OC\\Preview\\TXT\"","level":0,"time":"2017-03-13T09:15:10-07:00","method":"GET","url":"\/index.php\/core\/preview.png?file=%2Fxxxx%2Fyyyyy%2Fzzzzz%2F0000.txt&c=589c9d17172bc&x=32&y=32&forceIcon=0","user":"mbs"}
```

The new log entry should look like the following:
```
{"reqId":"Q71WK4Sp6B03QqpOoxit","level":0,"time":"2017-03-13T09:15:10-07:00","remoteAddr":"173.18.243.1","user":"mbs","app":"core","method":"GET","url":"\/index.php\/core\/preview.png?file=%2Fxxxxx%2Fyyyyy%2Fzzzz%2F0000.txt&c=589c9d17172bc&x=32&y=32&forceIcon=0","message":"Generating preview for \"\/xxxxx\/yyyyy\/zzzz\/0000.txt\" with \"OC\\Preview\\TXT\""}
```

As an additional change, the reqId should replace the `/` to any other alphanumeric character ( @DeepDiver1975 @PVince81  , this extra change should be harmless)

#### Reasons for this change

* The reqId should have a fixed length, so the vision isn't shifted. Right now this isn't always happening due to `/` being escaped. The intention is that this char doesn't appear so we don't need to escape.
* Log level and time should already be fixed length, so we just need to move them to the beginning of the string. The reason is the same as the previous point.
* The next elements are the remoteAddr, user and app. The remoteAddr should be the first of the bunch because it's the more "stable": length should be between 11, 13, and 15 chars, but the length difference should be minimum. The user is usually associated with an ip (obviously not always true), so it's better to keep them side by side. Finally, the app is the last of the bunch because while we can't assume any length, but it shouldn't grow like crazy.
* The method and url should also be side by side, being the method the first because we can't assume anything about the url.
* The message should be always the last.

#### Important note

The format of each entry will still be generated using json. This can't be changed because there could be apps relying on this format.

### PR DESCRIPTION
**Pull request by virajparimi**
## Description

I have reordered the log entries and removed `/` character from being used in general for `reqID`.

## Related Issue

Fixes [#27397](https://github.com/owncloud/core/issues/27397)

## How Has This Been Tested?

Tested on Chrome and Ubuntu 14.04

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Comment by PVince81** [comment]
@jvillafanez please review, this was your wish 😄

File: lib/private/Security/SecureRandom.php [hunk 1]
```diff
@@ -72,7 +72,7 @@ public function getMediumStrengthGenerator() {
 	 * @return string
 	 */
 	public function generate($length,
-							 $characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/') {
+							 $characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+') {
```
Comment (jvillafanez) [inline lib/private/Security/SecureRandom.php]: I'd keep it as it is. Changing it here could have more consecuences. I'd make the change in the `getId` function of the request.

**Comment by jvillafanez** [comment]
@viraj96 any update on this? Should I take over it? I don't want to ruin your contribution, but I'd like to have this for 10.0 and we're running out of time.

**Comment by viraj96** [comment]
@jvillafanez Sorry, I was swamped with university projects this week and had to meet those deadlines, so I was not able to contribute for past few days. I will start contributing in a day or two. Sorry for the delay and thanks for your patience.
PS: I did make those changes initially but backtracked those considering the ease of the above method. But now since you confirmed the requirement, I will go ahead and do them.

**Comment by jvillafanez** [comment]
Ok, no problem.

Just to make sure, the `getId` method of the request should have a call to the `SecureRandom->generate(....)` method. As you've seen, that method accepts a length and a string with characters. You just need to add the second parameter with the characters we want to use.

**Comment by jvillafanez** [comment]
@viraj96 I don't think we can delay this anymore if we want this to be in 10.0. I'll take over this tomorrow if there hasn't been any progress.

**Comment by jvillafanez** [comment]
Superseded by https://github.com/owncloud/core/issues/27562. Closing.

**Pull request closed without merging**